packages/lesswrong/server/resolvers/ultraFeedResolver.ts

```diff
@@ -64,85 +64,54 @@ export const ultraFeedGraphQLTypeDefs = gql`
   }
 `
 
+// items now carry `type`
+type SampledItem =
+  | { type: "feedCommentThread"; feedCommentThread: FeedCommentsThread }
+  | { type: "feedPostWithContents"; feedPost: FeedFullPost }
+  | { type: "feedPost"; feedPostStub: FeedPostStub }          // ← NEW stub
+  | { type: "feedSpotlight"; feedSpotlight: FeedSpotlight };
+
 interface WeightedSource {
   weight: number;
-  items: FeedItem[];
-  renderAsType: FeedItemRenderType;
+  items: SampledItem[];
 }
 
-type SampledItem = { renderAsType: "feedCommentThread", feedCommentThread: FeedCommentsThread }
-                 | { renderAsType: "feedPost", feedPost: FeedFullPost }
-                 | { renderAsType: "feedSpotlight", feedSpotlight: FeedSpotlight };
-
 const weightedSample = (
-  inputs: Record<FeedItemSourceType, WeightedSource>,
+  inputs: Partial<Record<FeedItemSourceType, WeightedSource>>,
```

Comment:
why did this need to change to partial?

Reply:
It returns an object with only keys for sources that have non-zero items (I guess before it returned empty ones), hence partial